src/main/java/org/jboss/qa/jenkins/test/executor/utils/SyncProcessRunner.java

```diff
@@ -0,0 +1,73 @@
+package org.jboss.qa.jenkins.test.executor.utils;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class SyncProcessRunner {
+
+	private final ProcessBuilder processBuilder;
+
+	public static int run(ProcessBuilder processBuilder) throws IOException, InterruptedException {
+		return new SyncProcessRunner(processBuilder).run();
+	}
+
+	public SyncProcessRunner(ProcessBuilder processBuilder) {
+		this.processBuilder = processBuilder;
+	}
+
+	public int run() throws IOException, InterruptedException {
+		final Process process = processBuilder.start();
+		final RedirectProcessOut out = new RedirectProcessOut(process, false);
+		out.start();
+		final RedirectProcessOut err = new RedirectProcessOut(process, true);
+		err.start();
+
+		final int result = process.waitFor();
+		out.join();
+		err.join();
+		return result;
+	}
+
+	/**
+	 * This thread is workaround, because processBuilder.redirectOutput(ProcessBuilder.Redirect.INHERIT)
+	 * doesn't work on HPUX.
+	 */
+	public class RedirectProcessOut extends Thread {
+
+		private final boolean forErr;
+		private final Process process;
+		private final PrintStream print;
+
+		public RedirectProcessOut(final Process process, final boolean forErr) {
```

Comment:
@mswiech please choose a better name for class, ProcessOutputConsumer or something like that.